dashboard/app/controllers/reference_guides_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ReferenceGuidesController < ApplicationController
+  include CurriculumHelper
+  authorize_resource
+
+  before_action :require_levelbuilder_mode_or_test_env
+
+  def show
+    if params[:course_course_name] && params[:key]
+      course_version_id = find_matching_course_version(params[:course_course_name])&.id
```

Comment:
If we don't find an course_version_id it would probably be good to have a helpful error message.